typing/typecore.ml

```diff
@@ -1085,13 +1085,18 @@ and type_pat_aux ~exception_allowed ~constrs ~labels ~no_existentials ~mode
       ~env
   in
   let loc = sp.ppat_loc in
+  let rup k x =
+    if constrs = None then (ignore (rp x));
```

Comment:
You record the type of the pattern before unification with the expected type.
This is a change w.r.t. the previous behavior.
Is this intentional?

Reply:
Sorry, looking at what does `Stypes.record`, you are just registering the node earlier, and unification can still occur afterwards. So one has more information if the next unification fails, and the same otherwise. Fine.